### PR TITLE
chore: Update gha checkout and node versions.

### PR DESCRIPTION
**Pull request by yusinto**
Update checkout and setup-node actions to v4. Update node versions to 20 which uses npm v10 which supports provenance.

Screenshot_2024-04-16_at_11_41_26 AM

### CODE REVIEW

File: actions/install-npm-version/action.yml
Comment (yusinto) [inline actions/install-npm-version/action.yml:1]: node v20 comes with npm v10 which supports provenance out of the box so we don't need this anymore.

File: .github/workflows/server-node.yml [hunk 1]
```diff
@@ -17,14 +17,14 @@ jobs:
     strategy:
       matrix:
         # Node versions to run on.
-        version: [16, 18]
+        version: [18, 20]
```
Comment (kinyoklion) [inline .github/workflows/server-node.yml:20]:
```suggestion
        version: [18, 24]
```

I think we should be able to do this now.

**Review by kinyoklion — APPROVED** [review]

**Comment by kinyoklion** [comment]
Is `ga` supposed to be `gh` in the title?

**Comment by yusinto** [comment]
> Is `ga` supposed to be `gh` in the title?

I was trying to say github-actions. I'll re-write to say gha to make it clearer and less confusing.

File: .github/workflows/server-node.yml [hunk 2]
```diff
@@ -17,14 +17,14 @@ jobs:
     strategy:
       matrix:
         # Node versions to run on.
-        version: [16, 18]
+        version: [18, 24]
```
Comment (kinyoklion) [inline .github/workflows/server-node.yml:20]:
```suggestion
        version: [18, 21]
```
I put the wrong version in my suggestion. Sorry about that. :pray:

**Pull request merged**